files/image_config/misc/docker-wait-any

```diff
@@ -14,39 +14,108 @@
     NOTE: This script is written against docker Python package 4.1.0. Newer
     versions of docker may have a different API.
 """
-
+import argparse
+import subprocess
 import sys
 import threading
+import time
+
 from docker import APIClient
+from sonic_py_common import logger
+from swsssdk import SonicV2Connector
+
+SYSLOG_IDENTIFIER = 'docker-wait-any'
+
+# Global logger instance
+log = logger.Logger(SYSLOG_IDENTIFIER)
 
 # Instantiate a global event to share among our threads
 g_thread_exit_event = threading.Event()
+g_service = []
+g_dep_services = []
+
+# Check if System warm reboot, or Container warm restart is enabled.
+def is_warm_restart_enabled(container_name):
+    state_db = SonicV2Connector(host='127.0.0.1')
+    state_db.connect(state_db.STATE_DB, False)
+
+    TABLE_NAME_SEPARATOR = '|'
+    prefix = 'WARM_RESTART_ENABLE_TABLE' + TABLE_NAME_SEPARATOR
+
+    # Get the system warm reboot enable state
+    _hash = '{}{}'.format(prefix, 'system')
+    wr_system_state = state_db.get(state_db.STATE_DB, _hash, "enable")
+    wr_enable_state = True if wr_system_state == "true" else False
 
+    # Get the container warm reboot enable state
+    _hash = '{}{}'.format(prefix, container_name)
+    wr_container_state = state_db.get(state_db.STATE_DB, _hash, "enable")
+    wr_enable_state |= True if wr_container_state == "true" else False
 
-def usage():
-    print("Usage: {} <container_name> [<container_name> ...]".format(sys.argv[0]))
-    sys.exit(1)
+    state_db.close(state_db.STATE_DB)
+    return wr_enable_state
 
+# Check if System fast reboot is enabled.
+def is_fast_reboot_enabled():
+    fb_system_state = 0
+    cmd = 'sonic-db-cli STATE_DB get "FAST_REBOOT|system"'
+    proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
```

Comment:
@judyjoseph why can't we just read from State DB same as is_warm_restart_enabled function. Do we need to do subprocess ?

Reply:
The way we write the FAST_REBOOT state is using the SET redis API **redis-cli -n 6 SET "FAST_REBOOT|system" "1"**. The dbconnector/interface.py API's available are all using redis **HGET** API's which  returns the value associated with field in the hash stored at key.

Hence had to use the different approach with sonic-db-cli to get FAST_REBOOT|system

Options, we can change FAST_REBOOT to also start using HGET/HSET, may be as a separate change ?

Reply:
I have added this request in the issue which Ying raised to track this cleanup item https://github.com/Azure/sonic-buildimage/issues/5690#issuecomment-714258980